src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts

```diff
@@ -504,18 +504,21 @@ export const useField = <T, FormType = FormData, I = T>(
       const { resetValue = true, defaultValue: updatedDefaultValue } = resetOptions;
 
       setPristine(true);
-      setIsModified(false);
-      setValidating(false);
-      setIsChangingValue(false);
-      setIsValidated(false);
-      setStateErrors([]);
-
-      if (resetValue) {
-        hasBeenReset.current = true;
-        const newValue = deserializeValue(updatedDefaultValue ?? defaultValue);
-        // updateStateIfMounted('value', newValue);
-        setValue(newValue);
-        return newValue;
+
+      if (isMounted.current) {
```

Comment:
I needed to add this `if` statement as there was a strange race condition in one of the tests.

Reply:
There is test coverage for `form.reset()` in the `use_form.test.tsx` which test this part of the code. Aside this is a very low risk change as we are simply protecting ourselves from updating the state on an unmounted component.